// actions/authenticate.js
import { LOG_IN_REQUEST,
    LOG_IN_SUCCESS,
    LOG_IN_ERROR,
    LOG_OUT_REQUEST,
    LOG_OUT_SUCCESS,
    LOG_IN_ADMIN} from 'actionTypes'

import { fetchUser, fetchAdmin} from 'api/index'
import {FETCH_BOOK_FAILURE, FETCH_BOOK_START, FETCH_BOOK_SUCCESS} from "actionTypes/.";


export const logInRequest = (email, password) => async dispatch => {
    dispatch({type: LOG_IN_REQUEST});

    const admin = await fetchAdmin(email, password);
    console.log("admin = " + admin);
    if(admin !== undefined) {
        dispatch({
            type: LOG_IN_ADMIN,
            payload: admin
        })
    }
    else {
        const user = await fetchUser(email, password);
        if (user !== undefined) {
            dispatch({
                type: LOG_IN_SUCCESS,
                payload: user
            })
        }
        else {
            dispatch({
                type: LOG_IN_ERROR
            })
        }
    }

};

//export function logInSuccess (user) {
    //return { type: LOG_IN_SUCCESS, payload: user };
//}
//export function logInError () {
    //return { type: LOG_IN_ERROR };
//}
export function logOutRequest () {
    return { type: LOG_OUT_REQUEST };
}
export function logOutSuccess () {
    return { type: LOG_OUT_SUCCESS };
}
