// containers/logIn/index.js
import React, {Component} from 'react'
import { connect } from 'react-redux'
import './logIn.css'
//import '../registr-auth-popup_media.css';
import {validateRegistrationLogIn} from '../../constants/validateConstants'
import { getUserIsLoadingState } from '../../userSelector'
import { logInRequest } from '../../actions/authenticate'
import PropTypes from 'prop-types'
import $ from "jquery"

const propTypes = {
    isLoading: PropTypes.bool.isRequired,
    logIn: PropTypes.func.isRequired
};

class AuthPopup extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            emailValid: false,
            passwordValid: false,
            formValid: false
        };
    }

    logInOnClick = (event) => {
        if (this.state.formValid) {
            this.props.logIn(this.state.email, this.state.password);
            this.setState({email: '', password: ''});
        }
        event.preventDefault();
    };

    handleChangeInput = (event) => {
        const name = event.target.name;
        const value = event.target.value;
        this.setState({[name]: value}, () => this.validateField(name, value));
    };

    validateField = (fieldName, value) => {
        let emailValid = this.state.emailValid;
        let passwordValid = this.state.passwordValid;
        switch (fieldName) {
            case 'email':
                emailValid = (validateRegistrationLogIn.email).test(value);
                break;
            case 'password':
                passwordValid = (validateRegistrationLogIn.password).test(value);
                break;
            default:
                break;
        }
        this.setState({
            emailValid,
            passwordValid,
            formValid: emailValid && passwordValid
        });
    };

    popupHide(event) {
        $(event.target).closest(".modal").find(".close").trigger("click");
    };

    render() {
        return (
            <div>
                <form>
                    <div className="modal fade auth-popup" id="auth-popup" tabIndex="-1" role="dialog">
                        <div className="modal-dialog" role="document">
                            <div className="modal-content">
                                <div className="modal-header">
                                    <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                                        <span aria-hidden="true">&times;</span>
                                    </button>
                                    <h4 className="modal-title">Авторизация</h4>
                                </div>
                                <div className="modal-body">
                                    <input type="email" className="form-control" placeholder="Ваш e-mail" name="email"
                                           maxLength={30} value={this.state.email} onChange={this.handleChangeInput}/>
                                    <input type="password" className="form-control" placeholder="Ваш пароль"
                                           name="password"
                                           maxLength={30} value={this.state.password}
                                           onChange={this.handleChangeInput}/>
                                </div>
                                <div className="modal-footer">
                                    <button type="button" className="btn btn-primary" disabled={this.props.isLoading}
                                            onClick={this.logInOnClick}>Авторизоваться
                                    </button>
                                </div>
                                <div className="modal-link">
                                    <button type="button" className="btn btn-link">Я забыл(а) пароль</button>
                                    <button type="button" className="btn btn-link" data-toggle="modal"
                                            data-target="#registr-popup"
                                            onClick={(e) => this.popupHide(e)}>Зарегистрироваться
                                    </button>
                                </div>
                            </div>
                        </div>
                    </div>
                </form>
            </div>

        );
    }
}

AuthPopup.propTypes = propTypes;

const mapStateToProps = (state) => {
    console.log(state);
    return {
        isLoading: getUserIsLoadingState(state)
    };
}

const mapDispatchToProps = (dispatch) => {
    return ({
        logIn: (login, password) => dispatch(logInRequest(login, password))
    });

}
console.log(mapDispatchToProps);
export default connect(mapStateToProps, mapDispatchToProps)(AuthPopup);
